jobs/integr8ly/ocp4/install/image-deploy/Jenkinsfile

```diff
@@ -0,0 +1,175 @@
+            }
+
+            stage ('Build and push new integreatly-image') {
+                dir('integreatly-operator') {
+                    withCredentials([string(credentialsId: "integreatly-quay-bot-password", variable: "PASSWORD")]) {
+                        sh "docker login -u=\"integreatly+intlytravis\" -p=\"${PASSWORD}\" quay.io"
+                        sh "make image/build/push ORG=integreatly TAG=nightly"
+                    }
+                }
+            }
+
+            stage ('Prepare cluster resources') {
+                // note. `make cluster/prepare/osrc` needs to be removed when 
+                dir('integreatly-operator') {
+                    sh(
+                        returnStdout: false,
+                        script: """
+                            make cluster/prepare/project
+                            make cluster/prepare/secrets
+                            make cluster/prepare/configmaps
+                            oc create -f deploy/crds/*_crd.yaml
+                            oc create -f deploy/service_account.yaml
+                            oc create -f deploy/role.yaml
+                            oc create -f deploy/role_binding.yaml
+                        """
+                    )
+                }
+            }
+
+            stage('Fetch and deploy operator.yaml template') {
+                sh "curl -O https://gitlab.cee.redhat.com/integreatly-qe/rhmi-jenkins/raw/master/templates/operator.yaml"
+                sh "oc create -f operator.yaml"
+            } // stage
+
+            stage ('Create Installation CR') {
+                dir('integreatly-operator') {
+                    sh(
+                        returnStdout: false,
+                        script: """
+                            sed -i "s@SELF_SIGNED_CERTS@${selfSignedCerts}@" deploy/crds/examples/integreatly-installation-cr.yaml
+                            sed -i "s@INSTALLATION_NAME@${installationName}@" deploy/crds/examples/integreatly-installation-cr.yaml
+                            sed -i "s@INSTALLATION_TYPE@${INSTALLATION_TYPE}@" deploy/crds/examples/integreatly-installation-cr.yaml
+                        """
+                    )
+
+                    sh "oc create -f deploy/crds/examples/integreatly-installation-cr.yaml -n ${installationNamespace}"
+                    sleep time: 2, unit: 'MINUTES'
+                }
+            } // stage
+
+            stage ('Verify Installation Has Finished') {
+                Map<String, Integer> stages = new LinkedHashMap<String, Integer>()
+                stages.put("bootstrap", 10)
+                stages.put("cloud-resources", 20)
+                stages.put("monitoring", 20)
+                stages.put("authentication", 20)
+                stages.put("products", 60)
+                stages.put("solution-explorer", 20)
+                String getStageStatusCommand = ""
+                
+                for (int i = 0; i < stages.size(); i++) {
```

Comment:
I think it would be more handy to use some different iteration here. E.g `for (entry in map) {` or something like that.